code/modules/atmospherics/environmental/LINDA_system.dm

```diff
@@ -53,27 +53,79 @@
 /atom/movable/proc/block_superconductivity() // objects that block air and don't let superconductivity act
 	return FALSE
 
-/turf/proc/immediate_calculate_adjacent_turfs()
+/// This proc is a more deeply optimized version of immediate_calculate_adjacent_turfs
+/// It contains dumbshit, and also stuff I just can't do at runtime
+/// If you're not editing behavior, just read that proc. It's less bad
+/turf/proc/init_immediate_calculate_adjacent_turfs()
 	//Basic optimization, if we can't share why bother asking other people ya feel?
+	// You know it's gonna be stupid when they include a unit test in the atmos code
+	// Yes, inlining the string concat does save 0.1 seconds
+	#ifdef UNIT_TESTS
+	ASSERT(UP == 16)
+	ASSERT(DOWN == 32)
+	#endif
+	LAZYINITLIST(src.atmos_adjacent_turfs)
+	var/list/atmos_adjacent_turfs = src.atmos_adjacent_turfs
+	var/canpass = CANATMOSPASS(src, src, FALSE)
+	// I am essentially inlineing two get_dir_multizs here, because they're way too slow on their own. I'm sorry brother
+	var/list/z_traits = SSmapping.multiz_levels[z]
+	for(var/direction in GLOB.cardinals_multiz)
+		// Yes this is a reimplementation of get_step_mutliz. It's faster tho. fuck you
+		var/turf/current_turf = (direction & (UP|DOWN)) ? \
+			(direction & UP) ? \
+				(z_traits["16"]) ? \
+					(get_step(locate(x, y, z + z_traits["16"]), NONE)) : \
```

Comment:
no need for the second z_traits read here since its always going to be TRUE if it exists

Reply:
also why the get_turf() around locate()?

Reply:
> no need for the second z_traits read here since its always going to be TRUE if it exists

It isn't. trait["UP"] is the AMOUNT to move up or down to land on the z level, not just IF the z level exists or not
WAIT A MOMENT IT IS WHAT THE FUCK WHY DOES THAT CODE EVEN EXIST IF IT'S NOT GONNA FUCKING WORK

KEVINNNNNNNNNNNNNNNNNNNNNN